### ISSUE TITLE
Remove plugin-specific node packages

### ISSUE THREAD

**Issue by cldwalker** [LightTable/LightTable#1956]
As noticed in #1781, the following node packages belong in their respective plugins:

* bencode - Clojure. Also check that minor change in atom-shell branch isn't needed
* harbor - Javascript
* jshint - JSHint. JSHint already comes with node package but verify it's being used

When adding a plugin-local install, be sure to require with something like: `(js/require (plugins/local-module "javascript" "acorn"))`

**Comment by cldwalker** [comment]
It seems we'll need to preserve the bencode.js tweak, https://github.com/LightTable/LightTable/commit/d951744c5832e444d782f777f001d8377a07cab0?diff=split#diff-8c4ea801ee96367cd1663a7d20843875. Without it, if I kick off a plugin clojurescript rebuild by saving nrepl.cljs, LightTable freezes

**Comment by kenny-evitt** [comment]
Should we request the Bencode change be merged upstream? Or is there something we can do to obviate the need for that change? GitHub code search definitely misses some results but even with Google only the Clojure plugin *nrepl.cljs* file seems to use the Bencode `decode` function.

**Comment by cldwalker** [comment]
I wouldn't recommend any upstream merge until we understand why it's needed. It's also possible that upgrading to the latest 0.7.0 bencode fixes that issue

**Comment by rundis** [comment]
tried latest from master with bencode 0.7.0 and couldn't get clojure projects or plugins to connect.
Pretty sure I got bencode 0.7.0 to work with the 0.8.0 release though. I'll give it a try again and see if that works (ie clojure plugin 0.2.0)

**Comment by rundis** [comment]
nah. must have done something weird last night. Not working with bencode 0.7.0. Nrepl is fired up, but it never completes the connection handshake.

**Comment by kenny-evitt** [comment]
@rundis Can you set a breakpoint for that line in the code and check the relevant values? It looks like it's handling null/empty/invalid values but, if so, it's mysterious to me that that's not already covered.

**Comment by cldwalker** [comment]
I've updated this as blocked until after the cljs upgrade since we may need to tweak bencode during the upgrade
